### PR TITLE
build.d: Add proper DDoc for all functions

### PR DESCRIPTION
**Pull request by JinShil**

### CODE REVIEW

File: src/build.d [hunk 1]
```diff
@@ -267,7 +284,15 @@ version(Windows)
     }
 }
 
-// Build individual CXX objects of the backend
+/**
+Gets the dependency that generates the given object file from the given source file
+
+Params:
+    obj      = the object file that the dependency should generate
+    filename = the source file to build, generating the object file
```
Comment (jacob-carlborg) [inline src/build.d]: The parameter name has a capital `N`.

File: src/build.d [hunk 2]
```diff
@@ -307,7 +340,9 @@ auto cxxBackend()
     return dependencies;
 }
 
-// Execute the sub-dependencies of the backend and pack everything into one object file
+/**
+Execute the sub-dependencies of the backend and pack everything into one object file
+*/
```
Comment (jacob-carlborg) [inline src/build.d]: Not sure what the style guide says but I would recommend three slashes instead.

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]

File: src/build.d [hunk 3]
```diff
@@ -159,7 +159,11 @@ auto newDelete()
     return dependency;
 }
 
-// Builds the lexer as a separate library
+/**
+Gets the dependency that builds the lexer
+
+Returns: the depedency that builds the lexer
```
Comment (Geod24) [inline src/build.d]: Usually for such trivial doc we only document the `Returns`.

File: src/build.d [hunk 4]
```diff
@@ -902,11 +913,7 @@ auto detectModel()
     throw new Exception(`Cannot figure 32/64 model from "` ~ uname ~ `"`);
 }
 
-/*
-Gets the command for querying or invoking the host C++ compiler
-
-Returns: the command for querying or invoking the host C++ compiler
-*/
+/// eturns: the command for querying or invoking the host C++ compiler
```
Comment (ZombineDev) [inline src/build.d]: Typo on "Returns".

File: src/build.d [hunk 5]
```diff
@@ -1030,31 +1033,52 @@ version(Windows)
...
     version(Windows)
         return name ~ ".dll";
     return name ~ ".a";
 }
 
-// Add additional make-like assignments to the environment
-// e.g. ./build.d ARGS=foo -> sets ARGS to 'foo'
+/**
+Add additional make-like assignments to the environment
+e.g. ./build.d ARGS=foo -> sets ARGS to 'foo'
```
Comment (ZombineDev) [inline src/build.d]: `sets the "ARGS" environment variable to "foo"`

File: src/build.d [hunk 6]
```diff
@@ -1030,31 +1033,52 @@ version(Windows)
...
+
+Params:
+    args = the assignments to add to the environment
```
Comment (ZombineDev) [inline src/build.d]: The command-line arguments from which the assignments will be parsed.

**Review by PetarKirov — APPROVED** [review]
Nice work overall. With a few minor nits fixed, feel free to auto-merge:

**Pull request merged**